tests/test_tpa_pipeline.py

```diff
@@ -124,30 +124,69 @@ def test_get_consent_status(self):
                 user=ec_user
             )
             assert get_consent_status_for_pipeline(pipeline) == consent
-            consent.delete()
-            ec_user.delete()
 
-    def test_handle_enterprise_logistration(self):
+    def test_handle_enterprise_logistration_no_pipeline(self):
         """
-        Test that we correctly handle incoming enterprise logistration requests.
+        Test that when there's no pipeline, we do nothing, then return.
         """
         backend = mock.MagicMock(name=None)
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
             fake_get_ec.return_value = None
             assert handle_enterprise_logistration(backend, self.user) is None
+            assert EnterpriseCustomerUser.objects.count() == 0
+
+    def test_handle_enterprise_logistration_consent_not_required(self):
+        """
+        Test that when consent isn't required, we create an EnterpriseCustomerUser, then return.
+        """
+        backend = mock.MagicMock(name=None)
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
             enterprise_customer = EnterpriseCustomerFactory(
                 enable_data_sharing_consent=False
             )
-            fake_get_ec.return_value = enterprise_customer  # pylint: disable=redefined-variable-type
+            fake_get_ec.return_value = enterprise_customer
             assert handle_enterprise_logistration(backend, self.user) is None
             assert EnterpriseCustomerUser.objects.filter(
                 enterprise_customer=enterprise_customer,
                 user_id=self.user.id
             ).count() == 1
+
+    def test_handle_enterprise_logistration_consent_required(self):
+        """
+        Test that when consent is required, we redirect to the consent page.
+        """
+        backend = mock.MagicMock(name=None)
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
-            fake_get_ec.return_value = self.customer  # pylint: disable=redefined-variable-type
+            fake_get_ec.return_value = self.customer
             assert isinstance(handle_enterprise_logistration(backend, self.user), HttpResponseRedirect)
+
+    def test_handle_enterprise_logistration_consent_optional(self):
+        """
+        Test that when consent is required, we redirect to the consent page.
+        """
+        backend = mock.MagicMock(name=None)
+        with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
+            self.customer.enforce_data_sharing_consent = EnterpriseCustomer.AT_ENROLLMENT
```

Comment:
nit: This test is called `test_handle_enterprise_logistration_consent_optional`, but it tests consent `AT_ENROLLMENT`.

Switch the name with [L173](https://github.com/edx/edx-enterprise/pull/53/commits/4ad7791f8ce300c5f9c0201b59018acacfcefa29#diff-e668e167d9d0bf0073a948bc1c155a90R173) below.